scikeras/wrappers.py

```diff
@@ -239,13 +243,252 @@ def _get_compile_kwargs(self):
+                and all(
+                    isinstance(el[0], str) or isinstance(el[0], None)
+                    for el in compile_kwargs["loss"]
+                )
+                and (
+                    all(
+                        isinstance(el[2], str) for el in compile_kwargs["loss"]
+                    )
+                    or all(
+                        isinstance(el[2], None)
+                        for el in compile_kwargs["loss"]
+                    )
+                )
+            ):
+                out = dict()
+                # Special format
+                for output_num, (name, loss, output_name) in enumerate(
+                    compile_kwargs["loss"]
+                ):
+                    output_name = output_name or output_num
+                    if isinstance(loss, str):
+                        loss = losses_module.get(loss)
+                    if isclass(loss):
+                        this_output_kwargs = loss_kwargs.copy()
+                        if name is not None:
+                            this_output_kwargs["name"] = name
+                            this_output_kwargs.update(
+                                route_params(
+                                    params,
+                                    destination=f"loss__{name}",
+                                    pass_filter=set(),
+                                    strict=True,
+                                )
+                            )
+                            this_output_kwargs.update(
+                                route_params(
+                                    params,
+                                    destination=f"compile__loss__{name}",
+                                    pass_filter=set(),
+                                    strict=True,
+                                )
+                            )
+                        out[output_name] = loss(**this_output_kwargs)
+                    else:
+                        out[output_name] = loss
+                if all(isinstance(key, int) for key in out.keys()):
+                    # Convert to list
+                    out = list(out.values())
+                compile_kwargs["loss"] = out
+            elif loss_kwargs and isclass(compile_kwargs["loss"]):
+                compile_kwargs["loss"] = compile_kwargs["loss"](**loss_kwargs)
+            elif loss_kwargs and isinstance(compile_kwargs["loss"], str):
+                compile_kwargs["loss"] = losses_module.get(
+                    compile_kwargs["loss"]
+                )
+                if hasattr(compile_kwargs["loss"], "name"):
+                    # got a class instance, get the class and re-compile it
+                    compile_kwargs["loss"] = compile_kwargs["loss"].__class__(
+                        **loss_kwargs
+                    )
+        # ------------------------------------------ METRICS ----------------------------------------------------
```

Comment:
Style nit and suggestion:

``` diff
- # ---------------------- OPTIMIZER -------------------------------
- # ---------------------- LOSS -------------------------------
- # ---------------------- METRICS -------------------------------
+ def _get_compile_optimizer(self, compile_kwargs):
+     ...
+ def _get_compile_loss(self, compile_kwargs):
+     ...
+ def _get_compile_metrics(self, compile_kwargs):
+     ...
  def _get_compile_kwargs(self):
      if "loss" in compile_kwargs:
+         compile_kwargs["loss"] = self._get_compile_loss(compile_kwargs)
```

Reply:
Thank you. A lot of style work needed. Btw, all of the non-commented out part is ~working, I just forgot to push the tests.

Reply:
Oh, don't rush. I saw this in my notifications, took a peek and though I might be able to offer a helpful comment.

Reply:
Change is implemented and tests are up.